code/game/objects/structures/docking_port.dm

```diff
@@ -232,3 +245,35 @@ var/global/list/all_docking_ports = list()
 	if(istype(D))
 		return D
 	return 0
+
+var/global/list/dockinglights = list()
+
+/obj/machinery/docklight
+	name = "docking light"
+	desc = "A light designed to warn of dangerous docking conditions. Exercise caution while flashing."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "docklight"
+	light_color = LIGHT_COLOR_ORANGE
+	machine_flags = MULTITOOL_MENU
+	var/triggered = 0
+	var/id_tag = "" //Mappers: This should match the areaname of the target destination port.
+	//Examples: "main research department", "research outpost", "deep space", "station auxillary docking", "north of the station", etc.
+
+/obj/machinery/docklight/New()
+	..()
+	dockinglights += src
```

Comment:
Do the opposite of this inside `Destroy()`

Reply:
Went ahead and committed this via the web app